File: backend/src/main/java/com/back/team11/domain/review/dto/ReviewResponseDto.java
package com.back.team11.domain.review.dto;

import com.back.team11.domain.review.entity.Review;

import java.time.LocalDateTime;

public record ReviewResponseDto(
        Long id,
        Long cafeId,
        Long memberId,
        String nickname,
        String content,
        LocalDateTime createdAt
) {
    public static ReviewResponseDto from(Review review) {
        return new ReviewResponseDto(
                review.getId(),
                review.getCafe().getId(),
                review.getMember().getId(),
                review.getMember().getNickname(),
                review.getContent(),
                review.getCreatedAt()
        );
    }
}

File: backend/src/main/java/com/back/team11/domain/review/dto/ReviewResponseDto.kt
package com.back.team11.domain.review.dto

import com.back.team11.domain.review.entity.Review
import java.time.LocalDateTime

data class ReviewResponseDto(
    val id: Long,
    val cafeId: Long,
    val memberId: Long,
    val nickname: String,
    val content: String,
    val createdAt: LocalDateTime
) {
    companion object {
        fun from(review: Review): ReviewResponseDto =
            ReviewResponseDto(
                review.id,
                review.cafe.id,
                review.member.id,
                review.member.nickname,
                review.content,
                review.createdAt!!
            )
    }
}
